Remove item and sprite images from runelite-client resources directory

Nightfirecat opened the pull request:
Continued from #4491.

Many plugins made use of image files which were exact copies (or close derivatives) of ingame item or sprite images. As these images can be generated using ItemManager and SpriteManager, or by making modifications to existing images, this PR seeks to remove as many of these extraneous images as possible.

In addition, the added `ImageUtil.java` file has also had tests added to cover nearly all functionality utilized in the new and existing methods added to the class.

After these commits, the images that remain are non-duplicated custom images with the following exceptions:

- resources/net/runelite/client/plugins/cluescrolls/emote.png; this is a close copy of the interface styles' 2010 emote menu tab icon, but is scaled differently and is centered within the image canvas
- resources/net/runelite/client/plugins/farmingtracker/farming.png; this is distinct from any existing farming icons I could find, and does not seem to be a simple resize of other farming icons
-  resources/net/runelite/client/plugins/poh/* and resources/net/runelite/client/plugins/worldmap/*; these images are all of different dimensions and makeups from spell icons, and seem to be created specially for the minimap and world map, so have been left unaltered
- skill_icons/* and skill_icons_small/*; skill_icons_small are copies of the images found on the online highscore webpage, and are not present in the client cache. skill_icons are found in the cache, but must be consumed by the highscores plugin which loads even when the client is outdated and cannot generate a SpriteManager to load from cache. Hence, both directories are necessary.
- resources/net/runelite/client/ui/components/**loading_spinner_darker.gif**; I'll be honest, I have no idea how well (or poorly) Java's awt handles modification of animated GIFs so I left this one alone

Nightfirecat commented:
@deathbeam I've fixed the getters and removed caching variables and extra clientThread calls. I'll add the image resource loader as a new commit shortly.

As discussed in Discord, I've left the screenshot plugin's clientThread call alone since it seems to be required.

Nightfirecat commented on runelite-client/src/main/java/net/runelite/client/game/ClanManager.java:
```diff
@@ -154,25 +150,18 @@ public void onClanChange(ClanChanged clanChanged)
-			modIconsLength = newModIcons.length;
-		}
-		catch (IOException e)
+		for (BufferedImage image : clanChatImages)
```
I'm not sure I can get rid of `clanChatImages` since it's used to implement `ClanManager.getClanImage`, but I definitely don't need to use two methods to implement this.

deathbeam replied:
ooh true, I always forget about that one. You can change that one to spriteManager.getSprite(CLANCHAT_IMAGES[.ordinal - 1], 0) i guess.

Nightfirecat replied:
Decided to keep `clanChatImages` around after all to avoid doing one resize/outline/toIndexedColor operation per call, which could be many per frame from the player indicators plugin. This loop will be over `CLANCHAT_IMAGES` instead as suggested.

Nightfirecat commented on runelite-client/src/main/java/net/runelite/client/plugins/cluescrolls/ClueScrollPlugin.java:
```diff
@@ -594,7 +591,29 @@ private void addMapPoints(WorldPoint... points)
+	{
+		try
+		{
+			synchronized (ImageIO.class)
```
I've added `ImageUtil::getResourceStreamFromClass` for this.

Nightfirecat commented on runelite-client/src/test/java/net/runelite/client/util/ImageUtilTest.java:
```diff
@@ -0,0 +1,437 @@
+	@Test
+	public void resizeImage()
+	{
+		// TODO: test image contents after changing size
```
The scaling algorithm does not exactly match that of other programs (like Photoshop) so I wasn't able to effectively test this. If needed, I could use the method to create some resized versions of a skill icon (like the overall skill icon) and add tests to compare them, but that seemed rather useless to me.

Nightfirecat commented:
Rebased and fixed a handful of UI images that needed adjusting. (Primarily `grayscaleOffset` -> `alphaOffset` for grayscale images after the "SwingUtil: Fix grayscaleOffset to not modify alpha" commit)

At this point, I've tested all of the color/alpha-modified UI images and found them to be exactly (or very nearly, in a few cases) identical to master's behavior. I've also been running this branch while doing normal gameplay and haven't noticed any issues; all replaced images seem to be loading without issue.

Nightfirecat commented:
Rebased.
